### DISCUSSION TITLE
Add to Benchmarks

### DISCUSSION THREAD

**Discussion by sirinath** [activej/activej#188] [Ideas]
Can you add this framework to benchmarks like:
- https://github.com/TechEmpower/FrameworkBenchmarks
- https://github.com/the-benchmarker/web-frameworks

**Comment by gembin** [comment]
For `core-net` and `core-http` any benchmarks comparing to https://github.com/netty/netty,  since netty is already widely used and proven for its performance.  Any reason to build `net` from scratch?

**Comment by sirinath** [comment]
Also there is https://github.com/OpenHFT/Chronicle-Network which is faster than Netty which can be bench-marked with. But Netty being so widely used definitely should be in the benchmarks.

**Comment by eduard-vasinskyi** [comment]
> Can you add this framework to benchmarks like:
> 
>     * https://github.com/TechEmpower/FrameworkBenchmarks
> 
>     * https://github.com/the-benchmarker/web-frameworks

@sirinath, we have added ActiveJ to both benchmarks.
ActiveJ has placed #&#8203;1 in [web-frameworks](https://github.com/the-benchmarker/web-frameworks) benchmark, significantly ahead of the rest!

As for TechEmpower's [FrameworkBenchmarks](https://github.com/TechEmpower/FrameworkBenchmarks), we have added ActiveJ for both JSON and PLAINTEXT benchmarks. We have had pretty good results when benchmarking our HTTP server locally.
Unfortunately, we were late for Round 20 of the benchmark. So, ActiveJ is probably going to be listed in the official results only when Round 21 comes.

**Comment by eduard-vasinskyi** [comment]
@gembin @sirinath
We could not find any public benchmarks for TCP servers. If you happen to know one, please let us know, we will consider adding ActiveJ to it.
As for netty, ActiveJ was first developed for internal projects back in the days when netty was not so mature as it is now. We have found its filter-based API to be not suitable for our programming model, so we have decided to build our own NIO server implementation from scratch.

**Comment by sirinath** [comment]
Maybe you can start your own benchmarks.

**Comment by eduard-vasinskyi** [comment]
Hello, @sirinath
We have a set of [benchmarks](https://github.com/activej/activej/tree/master/benchmarks) for our components, including benchmarks for TCP client and server from the 'net' module. 
We currently are not planning to maintain a public benchmark similar to the ones you have previously linked.

**Comment by sirinath** [comment]
There is a serialisation benchmarks: https://github.com/eishay/jvm-serializers

**Comment by eduard-vasinskyi** [comment]
Hello, @sirinath
We have forked this repo and added ActiveJ Serializer to it. Here is a [link](https://github.com/activej/jvm-serializers). The original repo does not seem to be maintained.

**Comment by sirinath** [comment]
Here is a stream processing benchmark: 
- https://github.com/tinyield/sequences-benchmarks, 
- https://github.com/tinyield/parallel-sequences-benchmarks, 
- https://github.com/tinyield/sequences-benchmarks-every-and-first
